### ISSUE TITLE
tuneParam() fails with cryptic error message

### ISSUE THREAD

**Issue by enricoferrero** [mlr-org/mlr#513]
I'm using tuneParams() to find an optimal percentage of features to keep on a large dataset (5000 observations, 25000 features).

I'm using R v3.2.0 and mlr v2.4 from CRAN.

I get this error message:
`Error in tuneParams(lrn, classif.task, rdesc, par.set = ps, control = makeTuneControlGrid()) :
  Assertion on 'y' failed: Must be of type 'numeric', not 'NULL'`

Here's a minimal reproducible example:
```{r}
library(mlr)
library(parallelMap)

set.seed(16)
parallelStart("multicore", 6)

dat <- data.frame(replicate(25000, sample(0:1, 5000, rep=TRUE)))
classif.task <- makeClassifTask(id="test", data=dat, target="X1")
lrn <- makeLearner("classif.svm")
lrn <- makeFilterWrapper(lrn, fw.method="information.gain")
rdesc <- makeResampleDesc("CV", iters=3)
ps <- makeParamSet(makeDiscreteParam("fw.perc", values = c(0.5, 0.25, 0.1, 0.05, 0.01, 0.001)))
res <- tuneParams(lrn, classif.task, rdesc, par.set=ps, control=makeTuneControlGrid())

parallelStop()
```

I'm guessing this might be related to the size of the data.frame, but I'm not sure.
Any idea on how I can fix this?
Thanks!

**Comment by enricoferrero** [comment]
Posting full output:

```{r}
> library(mlr)
> library(parallelMap)
> set.seed(16)
> parallelStart("multicore", 8)
Warning in parallelStart("multicore", 8) :
  Parallelization was not stopped, doing it now.
Stopped parallelization. All cleaned up.
Starting parallelization in mode=multicore with cpus=8.
> dat <- data.frame(replicate(25000, sample(0:1, 5000, rep=TRUE)))
classif.task <- makeClassifTask(id="test", data=dat, target="X1")
lrn <- makeLearner("classif.svm")
lrn <- makeFilterWrapper(lrn, fw.method="information.gain")
rdesc <- makeResampleDesc("CV", iters=3)
ps <- makeParamSet(makeDiscreteParam("fw.perc", values = c(0.5, 0.25, 0.1, 0.05, 0.01, 0.001)))
res <- tuneParams(lrn, classif.task, rdesc, par.set=ps, control=makeTuneControlGrid())
parallelStop()
> classif.task <- makeClassifTask(id="test", data=dat, target="X1")
> lrn <- makeLearner("classif.svm")
> lrn <- makeFilterWrapper(lrn, fw.method="information.gain")
> rdesc <- makeResampleDesc("CV", iters=3)
> ps <- makeParamSet(makeDiscreteParam("fw.perc", values = c(0.5, 0.25, 0.1, 0.05, 0.01, 0.001)))
> res <- tuneParams(lrn, classif.task, rdesc, par.set=ps, control=makeTuneControlGrid())
[Tune] Started tuning learner classif.svm.filtered for parameter set:
            Type len Def                       Constr Req Trafo
fw.perc discrete   -   - 0.5,0.25,0.1,0.05,0.01,0.001   -     -
With control class: TuneControlGrid
Imputation value: 1
Mapping in parallel: mode = multicore; cpus = 8; elements = 6.
[Tune-x] 1: fw.perc=0.5
[Tune-x] 2: fw.perc=0.25
[Tune-x] 3: fw.perc=0.1
[Tune-x] 4: fw.perc=0.05
[Tune-x] 5: fw.perc=0.01
[Tune-x] 6: fw.perc=0.001
Error in tuneParams(lrn, classif.task, rdesc, par.set = ps, control = makeTuneControlGrid()) :
  Assertion on 'y' failed: Must be of type 'numeric', not 'NULL'
> parallelStop()
Stopped parallelization. All cleaned up.
> traceback()
10: stop(simpleError(sprintf(msg, ...), call = sys.call(1L)))
9: mstop("Assertion on '%s' failed: %s", var.name, msg)
8: makeAssertion(res, vname(x, .var.name))
7: assertNumeric(y, len = length(op$y.names))
6: addOptPathEl.OptPathDF(opt.path, x = as.list(states[[i]]), y = res$y,
       exec.time = res$exec.time, error.message = res$errmsg, dob = dobs[i],
       eol = eols[i], check.feasible = FALSE, extra = extra)
5: addOptPathEl(opt.path, x = as.list(states[[i]]), y = res$y, exec.time = res$exec.time,
       error.message = res$errmsg, dob = dobs[i], eol = eols[i],
       check.feasible = FALSE, extra = extra)
4: evalOptimizationStates(learner, task, resampling, measures, par.set,
       NULL, control, opt.path, show.info, states, dobs, eols, remove.nas,
       "mlr.tuneParams")
3: evalOptimizationStatesTune(learner, task, resampling, measures,
       par.set, control, opt.path, show.info, xs, dobs = seq_along(xs),
       eols = NA_integer_, remove.nas = TRUE)
2: sel.func(learner, task, resampling, measures, par.set, control,
       opt.path, show.info)
1: tuneParams(lrn, classif.task, rdesc, par.set = ps, control = makeTuneControlGrid())
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.6 (Santiago)

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=C               LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=C              LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] parallelMap_1.3      mlr_2.4              ParamHelpers_1.5     ggplot2_1.0.1        BBmisc_1.9           BiocInstaller_1.18.4 vimcom_1.2-6         colorout_1.0-2       setwidth_1.0-4

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.1      magrittr_1.5     splines_3.2.0    MASS_7.3-44      munsell_0.4.2    colorspace_1.2-6 stringr_1.0.0    plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0       checkmate_1.6.2  gtable_0.1.2     e1071_1.6-7      class_7.3-14
[16] survival_2.38-3  digest_0.6.8     reshape2_1.4.1   stringi_0.5-5    scales_0.3.0     proto_0.3-10
```

Thanks!

**Comment by enricoferrero** [comment]
I can also reproduce it with a smaller dataset:

```{r}
library(mlr)
library(parallelMap)

set.seed(16)
parallelStart("multicore", 8)

dat <- data.frame(replicate(2500, sample(0:1, 500, rep=TRUE)))
classif.task <- makeClassifTask(id="test", data=dat, target="X1")
lrn <- makeLearner("classif.svm")
lrn <- makeFilterWrapper(lrn, fw.method="information.gain")
rdesc <- makeResampleDesc("CV", iters=3)
ps <- makeParamSet(makeDiscreteParam("fw.perc", values = c(0.5, 0.25, 0.1, 0.05, 0.01, 0.001)))
res <- tuneParams(lrn, classif.task, rdesc, par.set=ps, control=makeTuneControlGrid())

parallelStop()
```

```{r}
> library(mlr)
> library(parallelMap)
> set.seed(16)
> parallelStart("multicore", 8)
Warning in parallelStart("multicore", 8) :
  Parallelization was not stopped, doing it now.
Stopped parallelization. All cleaned up.
Starting parallelization in mode=multicore with cpus=8.
> dat <- data.frame(replicate(2500, sample(0:1, 500, rep=TRUE)))
> classif.task <- makeClassifTask(id="test", data=dat, target="X1")
> lrn <- makeLearner("classif.svm")
> lrn <- makeFilterWrapper(lrn, fw.method="information.gain")
> rdesc <- makeResampleDesc("CV", iters=3)
> ps <- makeParamSet(makeDiscreteParam("fw.perc", values = c(0.5, 0.25, 0.1, 0.05, 0.01, 0.001)))
> res <- tuneParams(lrn, classif.task, rdesc, par.set=ps, control=makeTuneControlGrid())
[Tune] Started tuning learner classif.svm.filtered for parameter set:
            Type len Def                       Constr Req Trafo
fw.perc discrete   -   - 0.5,0.25,0.1,0.05,0.01,0.001   -     -
With control class: TuneControlGrid
Imputation value: 1
Mapping in parallel: mode = multicore; cpus = 8; elements = 6.
[Tune-x] 1: fw.perc=0.5
[Tune-x] 6: fw.perc=0.001
[Tune-x] 5: fw.perc=0.01
[Tune-x] 2: fw.perc=0.25
[Tune-x] 4: fw.perc=0.05
[Tune-x] 3: fw.perc=0.1
Error in tuneParams(lrn, classif.task, rdesc, par.set = ps, control = makeTuneControlGrid()) :
  Assertion on 'y' failed: Must be of type 'numeric', not 'NULL'
> traceback()
10: stop(simpleError(sprintf(msg, ...), call = sys.call(1L)))
9: mstop("Assertion on '%s' failed: %s", var.name, msg)
8: makeAssertion(res, vname(x, .var.name))
7: assertNumeric(y, len = length(op$y.names))
6: addOptPathEl.OptPathDF(opt.path, x = as.list(states[[i]]), y = res$y,
       exec.time = res$exec.time, error.message = res$errmsg, dob = dobs[i],
       eol = eols[i], check.feasible = FALSE, extra = extra)
5: addOptPathEl(opt.path, x = as.list(states[[i]]), y = res$y, exec.time = res$exec.time,
       error.message = res$errmsg, dob = dobs[i], eol = eols[i],
       check.feasible = FALSE, extra = extra)
4: evalOptimizationStates(learner, task, resampling, measures, par.set,
       NULL, control, opt.path, show.info, states, dobs, eols, remove.nas,
       "mlr.tuneParams")
3: evalOptimizationStatesTune(learner, task, resampling, measures,
       par.set, control, opt.path, show.info, xs, dobs = seq_along(xs),
       eols = NA_integer_, remove.nas = TRUE)
2: sel.func(learner, task, resampling, measures, par.set, control,
       opt.path, show.info)
1: tuneParams(lrn, classif.task, rdesc, par.set = ps, control = makeTuneControlGrid())
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.6 (Santiago)

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=C               LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=C              LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] parallelMap_1.3      mlr_2.4              ParamHelpers_1.5     ggplot2_1.0.1        BBmisc_1.9           BiocInstaller_1.18.4 vimcom_1.2-6         colorout_1.0-2       setwidth_1.0-4

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.1      magrittr_1.5     splines_3.2.0    MASS_7.3-44      munsell_0.4.2    colorspace_1.2-6 stringr_1.0.0    plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0       checkmate_1.6.2  gtable_0.1.2     e1071_1.6-7      class_7.3-14
[16] survival_2.38-3  digest_0.6.8     reshape2_1.4.1   stringi_0.5-5    scales_0.3.0     proto_0.3-10
```

**Comment by larskotthoff** [comment]
I can't reproduce the error with R 3.2.2. Could you try with that version please?

**Comment by berndbischl** [comment]
duplicate of #514
